nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -185,56 +152,96 @@ def set_ee_ctrl_mode(self, mode):
+            logger_name=self.logger_name,
+        )
+        try:
+            response = lookup_client(RobotConfigurationLookup.LEFT_ARM, "home")
+            self.left_arm_joint_names = response.jnt_states.names
+            self.left_arm_home_joint_values = response.jnt_states.position
+
+            response = lookup_client(RobotConfigurationLookup.RIGHT_ARM, "home")
+            self.right_arm_joint_names = response.jnt_states.names
+            self.right_arm_home_joint_values = response.jnt_states.position
+
+            response = lookup_client(RobotConfigurationLookup.RIGHT_ARM, "handoff")
+            self.right_arm_handoff_joint_values = response.jnt_states.position
+
+            response = lookup_client(
+                RobotConfigurationLookup.LEFT_GRIPPER, "open_gripper"
+            )
+            self.left_gripper_joint_names = response.jnt_states.names
+            self.left_gripper_open_joint_values = response.jnt_states.position
+
+            response = lookup_client(
+                RobotConfigurationLookup.LEFT_GRIPPER, "closed_gripper"
+            )
+            self.left_gripper_closed_joint_values = response.jnt_states.position
+
+            response = lookup_client(
+                RobotConfigurationLookup.RIGHT_GRIPPER, "open_gripper"
+            )
+            self.right_gripper_joint_names = response.jnt_states.names
+            self.right_gripper_open_joint_values = response.jnt_states.position
+
+            response = lookup_client(
+                RobotConfigurationLookup.RIGHT_GRIPPER, "closed_gripper"
+            )
+            self.right_gripper_closed_joint_values = response.jnt_states.position
+
+        except rospy.ServiceException as exc:
+            rospy.logerr(
+                "Nightingale Manipulation Control failed to call robot configuration lookup service",
+                logger_name=self.logger_name,
+            )
+
+        self.joint_states_sub = rospy.Subscriber(
+            "/joint_states", JointState, self.update_joint_states
+        )
 
     def update_joint_states(self):
-        joint_states = rospy.client.wait_for_message("/joint_states", JointState)
-        p = joint_states.position
-        self.right_joint_states = [p[14], p[13], p[10], p[11], p[15], p[16], p[17]]
-        self.left_joint_states = [p[4], p[3], p[0], p[1], p[5], p[6], p[7]]
-        self.linear_joint_state = [p[8]]
-
-    def update_ee_pose(self):
-        """
-        robot_urdf = URDF.from_xml_string("urdf_str")
-        self.update_joint_states()
-        # base -> right ee
-        kdl_kin = KDLKinematics(robot_urdf, "base_link", "right_ee_link")
-        self.right_ee_pose = kdl_kin.forward(self.linear_joint_state + self.right_joint_states)
-
-        # base -> left ee
-        kdl_kin = KDLKinematics(robot_urdf, "base_link", "left_ee_link")
-        self.left_ee_pose = kdl_kin.forward(self.linear_joint_state + self.left_joint_states)
-        """
```

Comment:
I agree with this change for now, I am currently working on improving the update joint states and moving it into the joint control class.